### ISSUE TITLE
`disabled` in `MenuModel` not usable as a function

### ISSUE THREAD

**Issue by AndreasNasman** [primefaces/primevue#2396]
**I'm submitting a ...**
```
[x] bug report
```

**CodeSandbox Case**

See https://github.com/primefaces/primevue/issues/2365.

**Current behavior**
Using `disabled` as a function in `MenuModel` that returns `false` doesn't work as expected. The affected element is visually enabled (as it should) but doesn't respond to clicking, so it's still partly disabled.

**Expected behavior**
Setting the `disabled` property of `MenuModel` to a function returning `false` should completely enable an item, both visually and functionally wise.

**Minimal reproduction of the problem with instructions**
1. Add a `Menu` component and define an items array using the `MenuModel` structure.
1. Set one of the item's `disabled` values to a function returning `false`:
    - `disabled: () => false`
1. That item should now be enabled. However, it's only enabled visually. Clicking the item does nothing.

**What is the motivation / use case for changing the behavior?**
`disabled` should be usable as a function, as the documentation states.

**Please tell us about your environment:**
- macOS Monterey version 12.2.1
- Node version 14.19.0/16.14.0

* **Vue version:** 3.2.13

* **PrimeVue version:** 3.12.0

* **Browser:** all

**Comment by AndreasNasman** [comment]
This was already reported in #2365 but mistakenly closed by one of the maintainers.

**Comment by tugcekucukoglu** [comment]
Please follow the previous ticket. More than one ticket for the same issue is getting hard to follow the issue. 

Duplicate [#2365 ](https://github.com/primefaces/primevue/issues/2365)
